proof_of_stake/src/lib.rs

```diff
@@ -2423,3 +2418,141 @@ fn make_unbond_details<S>(
         slashed_amount,
     }
 }
+
+/// Tally a running sum of the fraction of rewards owed to each validator in
+/// the consensus set. This is used to keep track of the rewards due to each
+/// consensus validator over the lifetime of an epoch.
+pub fn log_block_rewards<S>(
+    storage: &mut S,
+    epoch: impl Into<Epoch>,
+    proposer_address: &Address,
+    votes: &[VoteInfo],
+) -> storage_api::Result<()>
+where
+    S: StorageRead + StorageWrite,
+{
+    // TODO: all values collected here need to be consistent with the same
+    // block that the voting info corresponds to, which is the
+    // previous block from the current one we are in.
+
+    // The votes correspond to the last committed block (n-1 if we are
+    // finalizing block n)
+
+    let epoch: Epoch = epoch.into();
+    let params = read_pos_params(storage)?;
+    let consensus_validators = consensus_validator_set_handle().at(&epoch);
+
+    // Get total stake of the consensus validator set
+    // TODO: this will need to account for rewards products?
```

Comment:
No issue needed, because this will already be solved in #1168. This won't actually be using the rewards products, but rather when the the inflation is determined for each validator, both the deltas and the validator set stakes will be updated directly.